File: Data Structures & Algorithms/subsets/submission-1.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> res;
        backtrack(0, {}, res, nums);
        return res;
    }
    void backtrack(int start, vector<int> subset, vector<vector<int>>& res, vector<int>& nums) {
        res.push_back(subset);
        for(int i = start; i < nums.size(); ++i) {
            subset.push_back(nums[i]);
            backtrack(i + 1, subset, res, nums);
            subset.pop_back();
        }
    }

};

File: Data Structures & Algorithms/pow-x-n/submission-1.py
class Solution:
    def myPow(self, x: float, n: int) -> float:

        if n == 1:
            return x
        elif n == -1:
            return 1 / x
        elif n == 0:
            return 1
        
        le = n // 2
        re = n - le
        print(le, re)
        return self.myPow(x, le) * self.myPow(x, re)